app/addons/fauxton/components.react.jsx

```diff
@@ -84,9 +84,129 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  var Tray = React.createClass({
+
+    getInitialState: function () {
+      return {
+        show: false
+      };
+    },
+
+    toggle: function (done) {
+      if (this.state.show) {
+        this.hide(done);
+      } else {
+        this.show(done);
+      }
+    },
+
+    componentDidMount: function () {
+      $('body').on('click.Tray-' + this.props.trayid + "-" + this._rootNodeID, function (e) {
```

Comment:
be careful regarding `this._rootNodeID`, it will break as soon as we update to 0.13:

from: https://facebook.github.io/react/blog/2015/03/10/react-v0.13.html

```
Access to most internal properties has been completely removed, including this._pendingState and this._rootNodeID.
```